opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -48,126 +52,104 @@ class Meter:
 
     def record_batch(
         self,
-        label_tuples: Dict[str, str],
-        record_tuples: List[Tuple["Metric", Union[float, int]]],
+        label_values: Tuple[str, ...],
+        record_tuples: Tuple[Tuple["Metric", ValueType]],
     ) -> None:
         """Atomically records a batch of `Metric` and value pairs.
 
         Allows the functionality of acting upon multiple metrics with
-        a single API call. Implementations should find handles that match
-        the key-value pairs in the label tuples.
+        a single API call. Implementations should find metric and handles that
+        match the key-value pairs in the label tuples.
 
         Args:
-            label_tuples: A collection of key value pairs that will be matched
-                against to record for the metric-handle that has those labels.
-            record_tuples: A list of pairs of `Metric` s and the
+            label_values: The values that will be matched against to record for
+                the handles under each metric that has those labels.
+            record_tuples: A tuple of pairs of `Metric` s and the
                 corresponding value to record for that metric.
         """
 
+    def create_metric(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: ValueType,
+        metric_kind: MetricKind,
```

Comment:
```suggestion
        metric_kind: type[_MetricT],
```

with `_MetricT` being a `TypeVar`. Then you can change the return value to `_MetricT` which enables you to eventually call `set` and `add` on the specific handles without having to fight mypy.

Reply:
`TypeVar` has some issues when converting the docs using Sphinx. I will used `Union` instead.

Reply:
That is not a replacement. It's impossible for mypy to deduce the return type with that (unless you think of something different than me). The docs can probably be fixed by using the `:param`/`:type` syntax if needs be.

Reply:
Good point. It seems `sphinx-autodoc-typehints` does not support TypeVar arguments. https://github.com/agronholm/sphinx-autodoc-typehints/issues/39

As a workaround, with using `:param`/`:type`, will the types need to be passed explicitly? I can't seem to get this to work with `:type value: _MetricT`